spec/rubocop/cop/lint/useless_assignment_spec.rb

```diff
@@ -940,6 +940,23 @@ def some_method(flag)
     end
   end
 
+  context "when there's an unreferenced assignment in top level if branch " \
+          'also assigned and referenced in the paired else branch in a block' \
+          'while the variable is referenced after the if/else' do
+    it 'allows' do
+      expect_no_offenses(<<~RUBY)
+        if cond
+          output = []
+        else
+          output = []
+          1.times { |value| output = [value] }
+        end
+
+        output
+      RUBY
+    end
+  end
```

Comment:
Can you add a test for numblocks too?

Reply:
@dvandersluis Added test for numblocks, updated changelog, and squashed it.